spring-kafka/src/main/java/org/springframework/kafka/listener/ConsumerAwareRebalanceListener.java

```diff
@@ -52,7 +52,7 @@ default void onPartitionsRevokedBeforeCommit(Consumer<?, ?> consumer, Collection
 			onPartitionsRevoked(partitions);
 		}
 		catch (Exception e) { // NOSONAR
-			LOGGER.debug(e, "User method threw exception");
+			LOGGER.error(e, "User method threw exception");
```

Comment:
>A callback interface that the user can implement to trigger custom actions when the set of partitions assigned to the consumer changes.
> (...)
> There are many uses for this functionality. One common use is saving offsets in a custom store. By saving offsets in the onPartitionsRevoked(Collection) call we can ensure that any time partition assignment changes the offset gets saved.
> Another use is flushing out any kind of cache of intermediate results the consumer may be keeping.

I can't imagine any scenario when I implement callback and is OK that framework silently ignores the exception thrown by  callback.
I always prefer rely on framework default when unexpected error appears rather than reinvent the wheel. That's idea of framework: focus on business and leave all rest default.
Swallow user exception is not good default in any framework

Reply:
OK. Let's hear then back from @garyrussell who did that logic in the `ConsumerAwareRebalanceListener`!

Reply:
I have no trouble with this; user listeners should never throw exceptions; this code was added "just in case".

Reply:
Just to clarify; this was also to protect against a strange use case where the user implements `ConsumerAwareRebalanceListener` but actually overrides one or more of the `ConsumerRebalanceListener` methods (with a `Consumer`).

https://github.com/artembilan/spring-kafka/commit/5879b41a8c410f156050c3549d7b61b17a636dac